lstore/lock.py

```diff
@@ -237,4 +241,13 @@ def upgrade_lock(self, transaction_id, record_id, current_lock_type, new_lock_ty
             else:
                 raise ValueError("Invalid lock upgrade")
             self.condition.notify_all()
+            
+    def has_lock(self, transaction_id, record_id, lock_type):
```

Comment:
This is extra check if you're using this you're not using wait-based 2PL with deadlock detection correctly